### PR TITLE
Benchmark multiply, division, modulo

### PR DESCRIPTION
**Pull request by arhimondr**

### CODE REVIEW

**Comment by pnowojski** [comment]
Please paste jmh results

**Comment by arhimondr** [comment]
```
MYLTIPLY OPERATOR

Fast Decimal
------------------------------------------

Benchmark                                         (expression)   Mode  Cnt  Score   Error   Units
BenchmarkDecimalOperators.myltiplyBenchmark            d1 * d2  thrpt   30  2.642 ± 0.012  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark  d1 * d2 * d3 * d4  thrpt   30  1.719 ± 0.008  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            i1 * i2  thrpt   30  2.774 ± 0.019  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            s1 * s2  thrpt   30  2.783 ± 0.044  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark  s1 * s2 * s5 * s6  thrpt   30  1.887 ± 0.014  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            s3 * s4  thrpt   30  0.451 ± 0.007  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            l2 * s2  thrpt   30  0.533 ± 0.015  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark  l2 * s2 * s5 * s6  thrpt   30  0.254 ± 0.006  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            s1 * l2  thrpt   30  0.510 ± 0.006  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            l1 * l2  thrpt   30  0.509 ± 0.017  ops/ms

Big Integer
------------------------------------------

Benchmark                                         (expression)   Mode  Cnt  Score    Error   Units
BenchmarkDecimalOperators.myltiplyBenchmark            s1 * s2  thrpt   30  2.817 ±  0.025  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark  s1 * s2 * s5 * s6  thrpt   30  1.884 ±  0.008  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            s3 * s4  thrpt   30  0.228 ±  0.012  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            l2 * s2  thrpt   30  0.164 ±  0.008  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark  l2 * s2 * s5 * s6  thrpt   30  0.058 ±  0.001  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            s1 * l2  thrpt   30  0.163 ±  0.002  ops/ms
BenchmarkDecimalOperators.myltiplyBenchmark            l1 * l2  thrpt   30  0.126 ±  0.002  ops/ms

DIVISION OPERATOR

Fast Decimal
------------------------------------------

Benchmark                                         (expression)   Mode  Cnt  Score    Error   Units
BenchmarkDecimalOperators.divisionBenchmark            d1 / d2  thrpt   30  2.627 ±  0.014  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  d1 / d2 / d3 / d4  thrpt   30  1.452 ±  0.015  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            i1 / i2  thrpt   30  1.439 ±  0.025  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  i1 / i2 / i3 / i4  thrpt   30  0.660 ±  0.002  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s1 / s2  thrpt   30  1.111 ±  0.021  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  s1 / s2 / s2 / s2  thrpt   30  0.478 ±  0.016  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s1 / s3  thrpt   30  0.190 ±  0.005  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s2 / l1  thrpt   30  0.270 ±  0.002  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            l1 / s2  thrpt   30  0.184 ±  0.003  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s3 / l1  thrpt   30  0.172 ±  0.004  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            l2 / l3  thrpt   30  0.139 ±  0.002  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  l2 / l4 / l4 / l4  thrpt   30  0.077 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  l2 / s4 / s4 / s4  thrpt   30  0.070 ±  0.001  ops/ms

Big Integer
------------------------------------------

Benchmark                                         (expression)   Mode  Cnt  Score    Error   Units
BenchmarkDecimalOperators.divisionBenchmark            s1 / s2  thrpt   30  0.978 ±  0.006  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  s1 / s2 / s2 / s2  thrpt   30  0.456 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s1 / s3  thrpt   30  0.086 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s2 / l1  thrpt   30  0.226 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            l1 / s2  thrpt   30  0.075 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            s3 / l1  thrpt   30  0.075 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark            l2 / l3  thrpt   30  0.052 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  l2 / l4 / l4 / l4  thrpt   30  0.034 ±  0.001  ops/ms
BenchmarkDecimalOperators.divisionBenchmark  l2 / s4 / s4 / s4  thrpt   30  0.024 ±  0.001  ops/ms

MODULO OPERATOR

Fast Decimal
------------------------------------------

Benchmark                                       (expression)   Mode  Cnt  Score   Error   Units
BenchmarkDecimalOperators.moduloBenchmark            d1 % d2  thrpt   30  0.415 ± 0.010  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  d1 % d2 % d3 % d4  thrpt   30  0.160 ± 0.001  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            i1 % i2  thrpt   30  1.465 ± 0.012  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  i1 % i2 % i3 % i4  thrpt   30  0.667 ± 0.011  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            s1 % s2  thrpt   30  0.332 ± 0.005  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  s1 % s2 % s2 % s2  thrpt   30  0.105 ± 0.006  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            s2 % l2  thrpt   30  0.317 ± 0.005  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            l3 % s3  thrpt   30  0.195 ± 0.001  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            s4 % l3  thrpt   30  0.214 ± 0.010  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            l2 % l3  thrpt   30  0.182 ± 0.006  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  l2 % l3 % l4 % l1  thrpt   30  0.069 ± 0.001  ops/ms

Big Integer
------------------------------------------

Benchmark                                       (expression)   Mode  Cnt  Score    Error   Units
BenchmarkDecimalOperators.moduloBenchmark            d1 % d2  thrpt   30  0.410 ±  0.007  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  d1 % d2 % d3 % d4  thrpt   30  0.154 ±  0.002  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            i1 % i2  thrpt   30  1.457 ±  0.064  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  i1 % i2 % i3 % i4  thrpt   30  0.604 ±  0.019  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            s1 % s2  thrpt   30  0.249 ±  0.019  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  s1 % s2 % s2 % s2  thrpt   30  0.091 ±  0.010  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            s2 % l2  thrpt   30  0.100 ±  0.008  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            l3 % s3  thrpt   30  0.052 ±  0.002  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            s4 % l3  thrpt   30  0.072 ±  0.003  ops/ms
BenchmarkDecimalOperators.moduloBenchmark            l2 % l3  thrpt   30  0.044 ±  0.001  ops/ms
BenchmarkDecimalOperators.moduloBenchmark  l2 % l3 % l4 % l1  thrpt   30  0.032 ±  0.003  ops/ms
```

**Comment by pnowojski** [comment]
lgtm

File: presto-main/src/test/java/com/facebook/presto/type/BenchmarkDecimalOperators.java [hunk 1]
```diff
@@ -443,7 +626,7 @@ public static void main(String[] args)
     {
         Options options = new OptionsBuilder()
                 .verbosity(VerboseMode.NORMAL)
-                .include(".*" + BenchmarkDecimalOperators.class.getSimpleName() + ".*")
+                .include(".*" + BenchmarkDecimalOperators.class.getSimpleName() + ".moduloBenchmark")
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/type/BenchmarkDecimalOperators.java:629]: undo

**Comment by sopel39** [comment]
lgtm % make sure all new code paths are tested

**Pull request merged**